coderd/audit.go

```diff
@@ -367,6 +367,26 @@ func (api *API) auditLogIsResourceDeleted(ctx context.Context, alog database.Get
 			api.Logger.Error(ctx, "unable to fetch workspace", slog.Error(err))
 		}
 		return workspace.Deleted
+	case database.ResourceTypeWorkspaceAgent:
+		// We use workspace as a proxy for workspace agents.
+		workspace, err := api.Database.GetWorkspaceByAgentID(ctx, alog.AuditLog.ResourceID)
+		if err != nil {
+			if xerrors.Is(err, sql.ErrNoRows) {
+				return true
+			}
+			api.Logger.Error(ctx, "unable to fetch workspace", slog.Error(err))
```

Comment:
Review: This code-path is confusing, I feel we should return an explicit boolean here, but I simply followed existing convention.

Reply:
I agree. Do we run a db query for every single audit log row? I was unaware this function existed tbh

Reply:
Yeah, sure looks like we call it as part of `convertAuditLog` 😰. I'm surprised the API response isn't slower TBH.